main/src/ca/uwaterloo/flix/language/phase/Verifier.scala

```diff
@@ -231,11 +231,39 @@ object Verifier {
             case _ => failMismatchedShape(t1, "Array", loc)
           }
 
-        case AtomicOp.StructNew(_, _) => throw new RuntimeException("JOE TBD")
+        case AtomicOp.StructNew(sym0, fields0) =>
+          val region :: fieldTpes = ts
```

Comment:
This pattern match could fail. I suppose thats still okay, but a neater structure might be:

ts match {
  case MonoType.Struct :: _ => // OK
  case _ => Error expected ...

Reply:
I would see if you can cleanup the code a bit :)